src/screens/UploadScreen.js

```diff
@@ -0,0 +1,198 @@
+import React, { Component } from "react";
+import {
+  View,
+  Text,
+  StyleSheet,
+  ScrollView,
+  TextInput,
+  Dimensions,
+  StatusBar,
+  ActivityIndicator,
+  Alert,
+  Modal,
+  SafeAreaView,
+  TouchableOpacity,
+} from "react-native";
+import { Video } from "expo-av";
+import VideoPlayer from "expo-video-player";
+import { Button } from "react-native-paper";
+import Colors from "../constants/Colors";
+
+const { width, height } = Dimensions.get("window");
+
+export default class UploadScreen extends Component {
+  state = {
+    title: "",
+    caption: "",
+    isSubmitting: false,
+    modal: false,
+  };
+
+  _onTitleChange = (text) => {
+    this.setState({
+      title: text,
+    });
+  };
+
+  _onCaptionChange = (text) => {
+    this.setState({
+      caption: text,
+    });
+  };
+
+  _submit = async () => {
+    const { title, caption } = this.state;
+    const { navigation } = this.props;
+
+    if (title && caption) {
+      this.setState({
+        isSubmitting: true,
+      });
+      const uploadResult = null;
+      if (!uploadResult) {
+        navigation.goBack(null);
+        navigation.goBack(null);
+        navigation.goBack(null);
```

Comment:
why does it get many null?

Reply:
Oh yes, thats weird haha because that way, it can go back to home screen without camera running in background. We will figure out and fix this in the future!

Reply:
Excellent! Did you implement that snippet yourself? I wonder if you can give me some references or docs.